libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/internal/impl/MapboxAudioGuidanceServices.kt

```diff
@@ -33,13 +35,17 @@ class MapboxAudioGuidanceServices {
         return MapboxSpeechApi(applicationContext, accessToken, language)
     }
 
-    fun mapboxVoiceInstructionsPlayer(
+    fun getOrUpdateMapboxVoiceInstructionsPlayer(
         mapboxNavigation: MapboxNavigation,
         language: String,
     ): MapboxVoiceInstructionsPlayer {
-        val applicationContext = mapboxNavigation.navigationOptions.applicationContext
-        val accessToken = mapboxNavigation.navigationOptions.accessToken!!
-        return MapboxVoiceInstructionsPlayer(applicationContext, accessToken, language)
+        return voiceInstructionsPlayer?.apply { updateLanguage(language) } ?: run {
```

Comment:
Wondering if we should remove the constructor entirely, turn this into a function. Because technically, accessToken changes are lost here

``` kotlin
private var voiceInstructionsPlayer = MapboxVoiceInstructionsPlayer()


...
fun updateMapboxVoiceInstructionsPlayer(mapboxNavigation, language) {
  val applicationContext = mapboxNavigation.navigationOptions.applicationContext
  val accessToken = mapboxNavigation.navigationOptions.accessToken!!
  voiceInstructionsPlayer.update(mapboxNavigation, language)
}
```

Reply:
also this class by the way, is a helper for tests (make it easier to mock interactions). any logic we add would be best inside a class so that it can be tested

Reply:
I was just looking into the accessToken.. I can't find where it is actually used inside, if it is not used; a comment here would be good

Reply:
or delete the`accessToken`

Reply:
I can't delete it because it's public API but I'll deprecate it.